### ISSUE TITLE
Do we have two factor Authentication in Orchard Core CMS ?

### ISSUE THREAD

**Issue by MuhammadTalha10** [OrchardCMS/OrchardCore#6352]

**Comment by sebastienros** [comment]
There isn't any default option right now.

I would suggest to follow the asp.net mvc implementation, and maybe this could be added as a PR?
Our User controller is following the default asp.net mvc templates, so it should be straightforward to add. And this could be done as a feature so it's not enabled by default. Each method of second auth could then be another feature. Like email, sms, ...

**Comment by MuhammadTalha10** [comment]
It may Included In future ? Or Orchard Core is not targeting to have this ?

**Comment by hishamco** [comment]
@MuhammadTalha10 I will try to add it in [OCC](https://github.com/OrchardCoreContrib/OrchardCoreContrib.Modules) in case OC doesn't include it yet

**Comment by MuhammadTalha10** [comment]
@sebastienros  @hishamco Thanks for your response

**Comment by Piedone** [comment]
There's a module for this by @ns8482e: https://github.com/surevelox/OrchardCore.Modules/blob/master/TwoFactorAuth/README.md

**Comment by Piedone** [comment]
@MikeAlhayek fixed in https://github.com/OrchardCMS/OrchardCore/pull/13704, right?

**Comment by hishamco** [comment]
Right

**Comment by Piedone** [comment]
Awesome!

**Comment by MikeAlhayek** [comment]
Yes this is now part of OC 1.7 preview. You can enable it easily via settings

https://github.com/OrchardCMS/OrchardCore/pull/13704

**Comment by Piedone** [comment]
Thank you for implementing it!

**Comment by hishamco** [comment]
I remembered @ns8482e implemented something similar previously, but it was closed-source, but @MikeAlhayek  did it on open :)

I might need to try what you did Mike, I use something similar to implement OTP with a custom SMS service

**Comment by MikeAlhayek** [comment]
Yes try it out :). I am already using it is production system. Yes with preview-1.7

![image](https://github.com/OrchardCMS/OrchardCore/assets/24724371/ace8631b-6498-4fd3-a8f6-8c1b1a8b507b)
